# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-am-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.15)
aclocal; autoheader; autoconf; automake -a -c
./configure --prefix=

# build
make

# install
install -d $DESTDIR/bin
install src/patch $DESTDIR/bin

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/patch-build.pc << EOF
Name: patch
Version: 2.7.6
Description: GNU patch
Requires: $REQUIRES

exec=\\
/bin/patch
EOF

exit

# patch
--- patch-2.7.6.orig/configure.ac
+++ patch-2.7.6/configure.ac
@@ -36,6 +36,8 @@
 AM_INIT_AUTOMAKE([1.11.2 -Wall gnu dist-bzip2 dist-xz color-tests parallel-tests])
 AM_SILENT_RULES([yes])
 
+AM_MAINTAINER_MODE
+
 AC_CONFIG_HEADERS([config.h:config.hin])
 
 AC_PROG_CC
--- patch-2.7.6.orig/src/pch.c
+++ patch-2.7.6/src/pch.c
@@ -241,8 +241,6 @@
 	  say (p_base
 	       ? "  Ignoring the trailing garbage.\ndone\n"
 	       : "  I can't seem to find a patch in there anywhere.\n");
-	if (! p_base && p_filesize)
-	  fatal ("Only garbage was found in the patch input.");
 	return false;
     }
     if (skip_rest_of_patch)

# source
https://ftp.gnu.org/gnu/patch/patch-2.7.6.tar.gz
